### PR TITLE
Implementation of bigquery-http4s

### PR DESCRIPTION
**Pull request by hamnis**
Implements Http4s client

This now works with [http4s-netty-client](https://mvnrepository.com/artifact/org.http4s/http4s-netty-client_3/0.5.14), so you should add that as a dependency when using bigquery-http4s

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -2,3 +2,4 @@ addSbtPlugin("org.typelevel" % "sbt-typelevel" % "0.6.5")
 addSbtPlugin("org.typelevel" % "sbt-typelevel-site" % "0.6.5")
 addSbtPlugin("org.scala-js" % "sbt-scalajs" % "1.15.0")
 addSbtPlugin("org.scalameta" % "sbt-mdoc" % "2.5.2")
+addSbtPlugin("net.hamnaberg" % "google-discovery-sbt" % "0.3.1")
```
Comment (hamnis) [inline project/plugins.sbt:5]: remove this

File: core/src/main/scala/no/nrk/bigquery/BQAdminClient.scala [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  def routinesInDataset(dataset: BQDataset.Ref): F[Vector[BQPersistentRoutine.Unknown]]
+
+  private def getExisting[A, B](id: A, get: A => F[Option[B]], name: String)(implicit M: MonadThrow[F]): F[B] =
+    get(id).flatMap {
+      case None =>
+        M.raiseError(new IllegalArgumentException(s"$name $id does not exists"))
+      case Some(b) => M.pure(b)
+    }
+
+  def createTempTable[Param](
```
Comment (hamnis) [inline core/src/main/scala/no/nrk/bigquery/BQAdminClient.scala:57]: Perhaps this should be on QueryClient

File: core/src/main/scala/no/nrk/bigquery/BQAdminClient.scala [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      table: BQTableDef.Table[Param],
+      tmpDataset: BQDataset.Ref
+  ): F[BQTableDef.Table[Param]] =
+    createTempTable(table, tmpDataset, Some(1.hour))
+
+  def createTempTable[Param](
```
Comment (hamnis) [inline core/src/main/scala/no/nrk/bigquery/BQAdminClient.scala:63]: Perhaps this should be on QueryClient

File: core/src/main/scala/no/nrk/bigquery/BQAdminClient.scala [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      table: BQTableDef.Table[Param],
+      tmpDataset: BQDataset.Ref,
+      expirationDuration: Option[FiniteDuration]
+  ): F[BQTableDef.Table[Param]]
+
+  def createTempTableResource[Param](table: BQTableDef.Table[Param], tmpDataset: BQDataset.Ref)(implicit
```
Comment (hamnis) [inline core/src/main/scala/no/nrk/bigquery/BQAdminClient.scala:69]: Perhaps this should be on QueryClient

File: build.sbt [hunk 5]
```diff
@@ -129,6 +139,29 @@ lazy val `google-client` = crossProject(JVMPlatform)
...
+  .dependsOn(core)
+  .settings(commonSettings)
+  .settings(
+    resolvers ++= Resolver.sonatypeOssRepos("snapshots"),
+    name := "bigquery-http4s-client",
+    libraryDependencies ++= Seq(
+      "org.scalameta" %% "munit-scalacheck" % "0.7.29" % Test,
+      "org.typelevel" %% "munit-cats-effect-3" % "1.0.7" % Test,
+      "io.chrisdavenport" %% "http4s-grpc-google-cloud-bigquerystorage-v1" % "3.0.0+0.0.6",
+      "net.hamnaberg.googleapis" %% "googleapis-http4s-bigquery" % "0.3-a13d31c-v2-20240214-SNAPSHOT",
```
Comment (hamnis) [inline build.sbt:155]: need release version

File: build.sbt [hunk 6]
```diff
@@ -136,6 +146,39 @@ lazy val `google-client` = crossProject(JVMPlatform)
...
+    resolvers ++= Resolver.sonatypeOssRepos("snapshots"),
+    name := "bigquery-http4s-client",
+    libraryDependencies ++= {
+      val binaryVersion = scalaBinaryVersion.value
+      Seq(
+        "org.scalameta" %% "munit-scalacheck" % "0.7.29" % Test,
+        "org.typelevel" %% "munit-cats-effect-3" % "1.0.7" % Test,
+        ("io.chrisdavenport" %% "http4s-grpc-google-cloud-bigquerystorage-v1" % "3.0.0+0.0.6")
+          .exclude("io.chrisdavenport", s"http4s-grpc_${binaryVersion}"),
+        ("io.chrisdavenport" %% "http4s-grpc" % "0.0.4")
```
Comment (hamnis) [inline build.sbt:165]: this has now been moved to the http4s organization. We should get a new release of that before merge

**Review by ingarabr — APPROVED** [review]

**Pull request merged**